src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java/JavaLanguageParser.g4

```diff
@@ -39,6 +39,24 @@ options { tokenVocab=JavaLanguageLexer; }
     private boolean isYieldStatement() {
         return _input.LT(1).getType() == JavaLanguageLexer.LITERAL_YIELD && switchBlockDepth > 0;
     }
+
+    static int fileCounter = 0;
+
+    /**
+     * We create a custom constructor so that we can clear the DFA
+     * states upon instantiation of JavaLanguageParser; this requires
+     * adding a dummy argument to constructor parameters.
+     *
+     * @param input the token stream to parse
+     * @param dummyArg dummy argument
+     */
+    public JavaLanguageParser(TokenStream input, boolean dummyArg) {
```

Comment:
Do we need to make similar changes to the Javadoc? Why?

Reply:
> Do we need to make similar changes to the Javadoc?

Assuming you mean Javadoc grammar here, we certainly could if we needed to; since this issue was reported for 9.X update, I only targeted the Java grammar.

Reply:
> Assuming you mean Javadoc grammar 

Yes.

> if we needed to

This is what my question is connected to. If we need further analysis on if Javadoc grammar needs this, then lets create an issue for it it. If there is no harm to adding it just to be sure we don't have to worry about it later, then I am fine with doing it too.

Reply:
After thinking on this, I believe that this may hurt Javadoc grammar performance more than helping it. Consider post at http://roman-ivanov.blogspot.com/2016/02/performance-problem-while-using-checks.html, where we can see that Javadoc parsing slows Checkstyle execution significantly. In this case, DFA clearing will likely slow parsing even more. I can open an issue if you would like, but the Javadoc grammar has much more severe issues than memory usage; I would address them before employing any memory hacks.

Reply:
I am ok if you are since you know this stuff more. I just want to make sure it was looked at.

Reply:
@nmancus1 , please update issue description (if we have issue to this) to look at upadate like in this PR to possibly improve javadoc speed.

update of javadoc is not in scope of this PR.

Reply:
https://github.com/checkstyle/checkstyle/issues/11193#issuecomment-1012809298